packages/sdk/src/evm/core/classes/contract-deployer.ts

```diff
@@ -1034,4 +1091,257 @@ export class ContractDeployer extends RPCConnectionHandler {
+            encodedArgs: encodedArgs,
+          };
+        } else {
+          this.infraContractsInfoCache[chainId] = {
+            [contract.contractType]: {
+              predictedAddress: address,
+              bytecode: metadata.compilerMetadata.bytecode,
+              encodedArgs: encodedArgs,
+            },
+          };
+        }
+      }
+    }
+
+    return this.infraContractsInfoCache[chainId][contract.contractType]
+      .predictedAddress;
+  }
+
+  private async computeCreate2FactoryAddress() {
+    const chainId = (await this.getProvider().getNetwork()).chainId;
+
+    if (
+      !this.infraContractsInfoCache[chainId] ||
+      !this.infraContractsInfoCache[chainId]["create2Factory"]
+    ) {
+      const enforceEip155 = await isEIP155Enforced(this.getProvider());
+
+      const tempChainId = enforceEip155 ? chainId : 0;
+      const deploymentInfo = getCreate2FactoryDeploymentInfo(tempChainId);
+
+      this.infraContractsInfoCache[chainId] = {
+        ...this.infraContractsInfoCache[chainId],
+        create2Factory: {
+          predictedAddress: deploymentInfo.deployment,
+          bytecode: "",
+          encodedArgs: [],
+        },
+      };
+    }
+
+    return this.infraContractsInfoCache[chainId]["create2Factory"]
+      .predictedAddress;
+  }
+
+  /**
+   * @internal
+   *
+   * Returns txn data for keyless deploys as well as signer deploys.
+   * Also provides a list of infra contracts to deploy.
+   *
+   * @param metadataUri
+   * @param activeChainId
+   * @param storage
+   */
+  async getDeploymentInfo(
+    metadataUri: string,
+    activeChainId?: number,
+  ): Promise<DeploymentInfo> {
+    const compilerMetadata = await fetchPreDeployMetadata(
+      metadataUri,
+      this.storage,
+    );
+    const chainId = activeChainId
+      ? activeChainId
+      : (await this.getProvider().getNetwork()).chainId;
+
+    // 1.  Get abi-encoded args and list of infra contracts required based on constructor params
+    const { encodedArgs, infraContracts } =
+      await this.encodeConstructorParamsForImplementation(
+        compilerMetadata,
+        chainId,
+      );
+
+    // 2. Compute the CREATE2 address
+    const predictedAddress = await computeDeploymentAddress(
+      compilerMetadata.bytecode,
+      encodedArgs,
+      await this.computeCreate2FactoryAddress(),
+    );
+
+    // 3. Add TWStatelessFactory and Forwarder to the list of infra contracts --
+    // these must be deployed regardless of constructor params of implementation contract
+    infraContracts.push(CloneFactory.contractType);
+
+    return {
+      bytecode: compilerMetadata.bytecode,
+      encodedArgs: encodedArgs,
+      predictedAddress: predictedAddress,
+      infraContractsToDeploy: infraContracts,
+    };
+  }
+
+  /**
+   * @internal
+   *
+   * Determine constructor params required by an implementation contract.
+   * Return abi-encoded params.
+   */
+  async encodeConstructorParamsForImplementation(
```

Comment:
i would put all these new functions in the `any-evm-utils.ts` file, or a new file. There's so much already in this class.